### ISSUE TITLE
Clean-up, esp. in documentation

### ISSUE THREAD

**Issue by a1russell** [kclay/rethink-scala#27]
Hi,

I've been considering trying out RethinkDB. Since Scala is my favorite language, I'd naturally use the Scala driver. However, upon finding this project, I have some reservations about actually using it.

I know it sounds shallow, but I am sure that if I have these reservations, other people will, too. I hope that my honesty does not offend you. Basically, there is a lot of stylistic sloppiness, even in the README file. For example, looking at the "to get started" section, there are a couple of lines commented out. What do they mean, and why are they commented out? Can we just delete them?

There is also a lot of whitespace usage inconsistency that makes the code simply difficult to read.

Looking through the code, I see that these minor issues are not isolated to the README.

Would you be opposed to me (or anyone, really) cleaning some of these things up?

**Comment by kclay** [comment]
Have at it. Never thought anyone would use this library. And no it doesn't sound shallow, I actually wanted to clean it up a bit but was never of importance for me.

**Comment by a1russell** [comment]
Great; thanks!

**Comment by ajrnz** [comment]
I like scala as well and RethinkDB interests me. I think it's quite likely that this library would be used however at the moment there is quite a barrier to entry. I've spent a couple of hours playing with it but can't get it to work either with the dependency 0.4.6 or compiling it myself. I'm sure it's just a question of imports or the examples being out of date but these thinks matter when you are trying to get started. The lack of contact information makes it very difficult to request help.

I'd appreciate the examples being a full scala file (with imports!) which just does some simple things like reading and writing data. If it was part of the project it would always compile.

Happy to help out if I can.

**Comment by kclay** [comment]
@ajrnz completely understand. I have some time today so I'll be creating a wiki so far I have it broken down to these topics:

1. Getting Started (sbt repo, imports, connection modes (blocking/async)
2. Type Safety ( some pitfalls/unsupported and converting between types)
3. Serialization (How it works and how to add custom parsers)
4. Sequences/Streams (map/contactMap, how to add more support if needed)
4. Querying (read/write, what the Document class is for, working with responses, lazy evaluations, future improvements)
5. Differences in api calls from rethinkdb.com/api (how to work with indexes, term options)

anything else I'm over looking?

**Comment by ajrnz** [comment]
No, that sounds good. I'll give you some more feedback when I've had a look. Thanks.

**Comment by fingerco** [comment]
I would like to request an example project or a little more thorough documentation of the methods and how to use them. I might be missing something obvious, but I can't quite figure out how to take a case class, insert it into a table and then get(key) to get it back.

**Comment by kclay** [comment]
@fingerco https://github.com/kclay/rethink-scala/blob/master/core/src/test/scala/example/blocking/Tutorial.scala

Also I'll make an issuse for the "db" bug. also its
r.db("myDB").table[MyClass]

**Comment by fingerco** [comment]
Thank you very much! This is very helpful.
